cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -278,7 +288,8 @@ class GcsIntegrationTest : public ::testing::Test {
 class TestGCSFSGeneric : public GcsIntegrationTest, public GenericFileSystemTest {
  public:
   void SetUp() override {
-    GcsIntegrationTest::SetUp();
+    GcsIntegrationTest::SetUp(); // Setup failure is not propagating
+    ASSERT_TRUE(Testbench()->running());
```

Comment:
One problem is the `SetUp()` assertion failures are not propagating to the child. The method just exits early and then the outer `SetUp()` continues as normal.

Reply:
Anyone else experienced this? Not sure what to do other than maybe move the contents of `SetUp()` into a separate function.